nl_server/tests/custom_embeddings_test.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Custom Embeddings."""
+
+from enum import Enum
+import os
+import shutil
+import unittest
+
+from parameterized import parameterized
+
+from nl_server import config
+from nl_server import embeddings
+from nl_server import embeddings_store as store
+from nl_server import gcs
+from shared.lib.gcs import TEMP_DIR
+
+_test_data = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                          'test_data')
+
+_DEFAULT_FILE: str = 'default.ft_final_v20230717230459.all-MiniLM-L6-v2.csv'
+_CUSTOM_FILE: str = 'custom.ft_final_v20230717230459.all-MiniLM-L6-v2.csv'
+_TUNED_MODEL: str = 'ft_final_v20230717230459.all-MiniLM-L6-v2'
+
+
+class DCType(str, Enum):
+  MAIN = "main"
+  CUSTOM = "custom"
+
+
+def _copy(fname: str):
+  dst = os.path.join(TEMP_DIR, fname)
+  shutil.copy(os.path.join(_test_data, fname), dst)
+  return dst
+
+
+class TestEmbeddings(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(cls) -> None:
+    default_file = _copy(_DEFAULT_FILE)
+    custom_file = _copy(_CUSTOM_FILE)
+
+    cls.main = store.Store(
+        config.load({'medium_ft': default_file}, {'tuned_model': _TUNED_MODEL}))
+
+    cls.custom = store.Store(
+        config.load({
+            'medium_ft': default_file,
+            'custom_ft': custom_file,
+        }, {'tuned_model': _TUNED_MODEL}))
+
+  def test_entries(self):
+    self.assertEqual(1, len(self.main.get('medium_ft').dcids))
+    self.assertTrue(not self.main.get('custom_ft'))
+    self.assertEqual(2, len(self.custom.get('medium_ft').dcids))
```

Comment:
Is this 2 because it is default + custom concatenated?

Reply:
Yes, left a comment.